nncf/quantization/algorithms/accuracy_control/algorithm.py

```diff
@@ -0,0 +1,331 @@
+from nncf.quantization.algorithms.accuracy_control.ranking import find_groups_of_quantizers_to_rank
+from nncf.quantization.algorithms.accuracy_control.ranking import remove_group_of_quantizers_from_model
+
+
+def get_algo_backend(backend: BackendType) -> AccuracyControlAlgoBackend:
+    """
+    Returns backend for accuracy control algorithm.
+
+    :param backend: Backend.
+    :return: The backend for accuracy control algorithm.
+    """
+    if backend == BackendType.OPENVINO:
+        from nncf.quantization.algorithms.accuracy_control.openvino_backend import OVAccuracyControlAlgoBackend
+        return OVAccuracyControlAlgoBackend()
+
+    raise RuntimeError('Cannot create the backend for the accuracy control algorithm '
+                       f'because {backend} is not supported.')
+
+
+def common_quantize_with_accuracy_control(model: TModel,
+                                          calibration_dataset: Dataset,
+                                          validation_dataset: Dataset,
+                                          validation_fn: Callable[[Any, Iterable[Any]], float],
+                                          max_drop: float = 0.01,
+                                          preset: QuantizationPreset = QuantizationPreset.PERFORMANCE,
+                                          target_device: TargetDevice = TargetDevice.ANY,
+                                          subset_size: int = 300,
+                                          fast_bias_correction: bool = True,
+                                          model_type: Optional[ModelType] = None,
+                                          ignored_scope: Optional[IgnoredScope] = None) -> TModel:
+    """
+    Common implementation of the `nncf.quantize_with_accuracy_control()` method.
+    """
+    # Step 0: Quantize provided model.
+    quantized_model = quantize(model, calibration_dataset, preset, target_device, subset_size,
+                               fast_bias_correction, model_type, ignored_scope)
+
+    # Backends
+    backend = get_backend(model)
+    algo_backend = get_algo_backend(backend)
+
+    initial_metric = validation_fn(algo_backend.prepare_for_inference(model),
+                                   validation_dataset.get_data())
+    nncf_logger.info(f'Metric of initial model:   {initial_metric}')
+
+    quantized_metric = validation_fn(algo_backend.prepare_for_inference(quantized_model),
+                                     validation_dataset.get_data())
+    nncf_logger.info(f'Metric of quantized model: {quantized_metric}')
+
+    return restore_accuracy(model, initial_metric,
+                            quantized_model, quantized_metric,
+                            validation_dataset, validation_fn, max_drop)
+
+
+def _create_message(nodes: Iterable[NNCFNode]) -> str:
+    names = [f'\t{x.node_name}' for x in nodes]
+    return '\n'.join(names)
+
+
+# TODO(andrey-churkin): Should be removed when native implementation will become the main one.
```

Comment:
Could you implement 2 implementation of quantize_with_accuracy_control based on POT and NATIVE implementation of quantize() function to have a chance to start test it ASAP.

Reply:
Done